modules/solid_mechanics/src/materials/SolidModel.C

```diff
@@ -989,7 +994,10 @@ SolidModel::crackingStrainDirections()
   {
     // Adjust the elasticity matrix for cracking.  This must be used by the
     // constitutive law.
-    _local_elasticity_tensor->adjustForCracking( _crack_flags_local );
+    if (_compute_method >= 2)
+      _local_elasticity_tensor->adjustForCracking( _crack_flags_local, true);
+    else
+      _local_elasticity_tensor->adjustForCracking( _crack_flags_local, false);
```

Comment:
I do not like this design. Passing bool flags to methods is not a good pattern (their meaning is not clear when looking at the call (unlike python which has named parameters)). I would suggest adding a new method for the `true` case with an expressive name (`adjustForCrackingWithShearRetention`). That method could call `adjustForCracking` and then perform the necessary modifications on `_val`.